src/Stack/SDist.hs

```diff
@@ -246,6 +260,61 @@ dirsFromFiles dirs = Set.toAscList (Set.delete "." results)
       | Set.member x s = s
       | otherwise = go (Set.insert x s) (FP.takeDirectory x)
 
+-- | Check package in given tarball. This will log all warnings
+-- and will throw an exception in case of critical errors.
+--
+-- Note that we temporarily decompress the archive to analyze it.
+checkSDistTarball :: (MonadIO m, MonadMask m, MonadThrow m, MonadCatch m, MonadLogger m, MonadReader env m, HasEnvConfig env)
+  => Path Abs File -- ^ Absolute path to tarball
+  -> m ()
+checkSDistTarball tarball = withTempTarGzContents tarball $ \pkgDir' -> do
+    pkgDir  <- (pkgDir' </>) `liftM`
+        (parseRelDir . FP.takeBaseName . FP.takeBaseName . toFilePath $ tarball)
+    --               ^ drop ".tar"     ^ drop ".gz"
+    cabalfp <- getCabalFileName pkgDir
+    name    <- parsePackageNameFromFilePath cabalfp
+    config  <- getPackageConfig name
+    (gdesc, pkgDesc) <- readPackageDescriptionDir config pkgDir
+    $logInfo $ "Checking package '" <>
+               (T.pack . P.unPackageName . P.pkgName . PD.package $ pkgDesc) <>
```

Comment:
Good point!  I've resolved this with https://github.com/commercialhaskell/stack/commit/44da1c1bdfa3fb6d0f62a54f05859d80b0efd34b and opened https://github.com/commercialhaskell/stack/issues/1398